opm/autodiff/WellInterface_impl.hpp

```diff
@@ -687,36 +712,56 @@ namespace Opm
             switch (workover) {
                 case WellEcon::CON:
                 {
-                    const bool last_connection = std::get<1>(ratio_check_return);
-                    const int worst_offending_connection = std::get<2>(ratio_check_return);
-
-                    assert((worst_offending_connection >= 0) && (worst_offending_connection < number_of_perforations_));
+                    const int worst_offending_completion = std::get<1>(ratio_check_return);
+
+                    wellTestState.addClosedCompletion(well_name, worst_offending_completion, simulationTime);
+                    if (writeMessageToOPMLog) {
+                        if (worst_offending_completion < 0) {
```

Comment:
I need to extend the WECON, but I do not understand Under what circumstances `worst_offending_completion` can be smaller than zero here.  What is the following `if and esle` about?
@totto82 , do you have some ideas?

Reply:
I am not sure if that is the case any longer, but I think the completion ID when complump is not used used to be negative. If negative, the connection number is logged, if positive, the completion number is logged.

Reply:
>  but I think the completion ID when complump is not used used to be negative.

When we do not use COMPLUMP, every connection gets a unique id when declared in COMPDAT, based on declaration order. I think it is this way now.